hypothesis-python/RELEASE.rst

```diff
@@ -0,0 +1,5 @@
+RELEASE_TYPE: minor
+
+Implement support for parameterized numpy arrays, with scaffolding
+for extra modules to register for early type resolution (resolving
+composite types).
```

Comment:
Style guide for release notes:

- I like to assume that readers are smart, but haven't heard of <whatever I'm writing about> - so adding cross-references to the docs for `numpy.typing.{ArrayLike,DtypeLike,NDArray}` would be nice.  Thanks to [intersphinx](https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html) (😉), this can use standard cross-reference syntax!
- Avoid mentioning implementation details or non-public interfaces - in this case, early resolution and the scaffolding should not be end-user-visible.

(looks like [our guide doc](https://github.com/HypothesisWorks/hypothesis/blob/master/guides/documentation.rst#changelog-entries) could have this added, I'll do that soon)

Reply:
Done. I didn't manage to make the `numpy.typing` xrefs work I'm afraid, not sure if it's a missing intersphinx mapping or what

```
OK - :class:`numpy.ndarray[scalar type] <numpy.ndarray>`
fail :class:`numpy.typing.NDArray[scalar type] <numpy.typing.NDArray>`
```